src/lang/terms/is_a.pl

```diff
@@ -142,3 +142,43 @@
     unique_name(Type_IRI,Name);
     Name = IRI
   ).
+
+     /*******************************
+     *	    UNIT TESTS	     		    *
+     *******************************/
+
+:- begin_tripledb_tests(
+    lang_is_a,
+    'package://knowrob/owl/test/swrl.owl',
+    [ namespace('http://knowrob.org/kb/swrl_test#')
+    ]).
+
+test("ask and tell woman is a person") :-
+  assert_true(is_a(test:'Woman', test:'TestThing')),
+  assert_false(is_a(test:'Woman', test:'Person')),
+  assert_true(tell(is_a(test:'Woman', test:'Person'))),
+  assert_true(is_a(test:'Woman', test:'Person')).
+
+test("ask and tell instances of Rex") :-
+  assert_true(instance_of(test:'Rex', test:'Man')),
+  assert_false(instance_of(test:'Rex', dul:'Person')),
+  assert_true(tell(instance_of(test:'Rex', dul:'Person'))),
+  assert_true(instance_of(test:'Rex', dul:'Person')).
+
+test("ask and tell list instance of Greta") :-
+  assert_false(instance_of('Greta', [test:'Woman', dul:'Person'])),
+  assert_true(tell(instance_of('Greta', [test:'Woman', dul:'Person']))),
+  assert_true(ask(instance_of_all('Greta', [test:'Woman', dul:'Person']))),
+  assert_true(ask(instance_of('Greta', [test:'Woman', dul:'Person']))).
+
+test("ask and tell sub property of") :-
+  assert_true(subproperty_of(test:'hasParent', test:'hasAncestor')),
+  assert_false(subproperty_of(test:'hasBrother', test:'hasSibling')),
+  assert_true(tell(subproperty_of(test:'hasBrother', test:'hasSibling'))),
+  assert_true(subproperty_of(test:'hasBrother', test:'hasSibling')).
+
+test("ask A is a Person and expect instantiation error as is_a expects a ground variable") :-
```

Comment:
better try to have a bit more compact name for the testcase.
If you think needed, add a comment in addition.

Reply:
Sure